File: Frontend/src/app/Message.ts
export class Message {
    id: number;
    public title: string;
    sender: string;
    public date: string;
    public body: string;
    broadcast: boolean;
    public recipients: Array<string>;

    constructor(title: string = "" , body: string = "" , broadcast: boolean = false, id: number =  0, sender: string = "", date: string = ""){
        this.id = id;
        this.title = title;
        this.sender = sender;
        this.date = date;
        this.body = body;
        this.broadcast = broadcast;
        this.recipients = [];
    }

    public getJson() {
        return {
            title: this.title,
            body: this.body,
            broadcast: this.broadcast,
            recipients: this.recipients
        };
    }
}

File: Backend/Common/src/main/java/hu/engineroom/common/entity/message/Message.java
package hu.engineroom.common.entity.message;

import hu.engineroom.common.entity.BaseEntity;
import hu.engineroom.common.entity.user.User;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@Entity
public class Message extends BaseEntity {

    private String body;

    private String title;

    private Date date;

    @ManyToOne
    private User sender;

    private boolean broadcast;

    @ManyToMany(cascade = CascadeType.PERSIST)
    @JoinTable(
            name = "user_messages",
            joinColumns = @JoinColumn(name = "messageId", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "userId", referencedColumnName = "id"))
    private List<User> recipients;
}
